File: DFS/DFS.js
let adjList= [];
let visited = [];

function dfs(cur){
    visited[cur] = true;

    for(let next of adjList[cur]){
        if(!visited[next]){
            dfs(next);
        }
    }
}
function main(){
    const fs = require('fs');
    const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

    const [n, m] = input[0].split(' ').map(Number);

    adjList = new Array(n+1).fill().map(() => [])
    visited = new Array(n+1).fill(false);

    for(let i=1; i<=m; i++){
        const [u, v] = input[i].split(' ').map(Number);
        adjList[u].push(v);
        adjList[v].push(u);
    }

    const start = Number(input[m+1])
    dfs(start);
}

File: DFS/DFS.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<vector<int>> adjList;
vector<bool> visited;

void dfs(int cur) {
  visited[cur] = true;

  for (int next : adjList[cur]) {
    if (!visited[next]) {
      dfs(next);
    }
  }
}

int main() {
  int n, m;
  cin >> n >> m;

  adjList.resize(n + 1);
  visited.resize(n + 1, false);

  for (int i = 0; i < m; i++) {
    int u, v;
    cin >> u >> v;
    adjList[u].push_back(v);
    adjList[v].push_back(u);
  }

  int start;
  cin >> start;

  dfs(start);

  return 0;
}
